indica/workflow/base_workflow.py

```diff
@@ -0,0 +1,581 @@
+            " _calculate_asymmetry_other_z method"
+        )
+
+    def calculate_asymmetry_other_z(self) -> DataArray:
+        self._asymmetry_parameter_other_z.update()
+        return self.asymmetry_parameter_other_z
+
+    def _derive_asymmetry_from_density(self, density: DataArray) -> DataArray:
+        """
+        Derive an asymmetry parameter from comparison of lfs (theta=0) and hfs (theta=1)
+        impurity density data
+        """
+        lfs = self.n_high_z.sel({"theta": 0}, method="nearest", drop=True).copy()
+        hfs = self.n_high_z.sel({"theta": np.pi}, method="nearest", drop=True).copy()
+        asymmetry: DataArray = (lfs - hfs) / (lfs + hfs)
+        return asymmetry
+
+    @property
+    def derived_asymmetry_high_z(self) -> DataArray:
+        return self._derived_asymmetry_high_z.data
+
+    def _calculate_derived_asymmetry_high_z(self) -> DataArray:
+        return self._derive_asymmetry_from_density(density=self.n_high_z)
+
+    def calculate_derived_asymmetry_high_z(self) -> DataArray:
+        self._derived_asymmetry_high_z.update()
+        return self.derived_asymmetry_high_z
+
+    @property
+    def derived_asymmetry_other_z(self) -> DataArray:
+        return self._derived_asymmetry_other_z.data
+
+    def _calculate_derived_asymmetry_other_z(self) -> DataArray:
+        return self._derive_asymmetry_from_density(density=self.n_other_z)
+
+    def calculate_derived_asymmetry_other_z(self) -> DataArray:
+        self._derived_asymmetry_other_z.update()
+        return self.derived_asymmetry_other_z
+
+    @property
+    def n_high_z(self) -> DataArray:
+        return self._n_high_z.data
+
+    @n_high_z.setter
+    def n_high_z(self, data: DataArray) -> None:
+        self._n_high_z.data = data
+
+    def _calculate_n_high_z(self) -> DataArray:
+        raise NotImplementedError(
+            f"{self.__class__} does not implement _calculate_n_high_z method"
+        )
+
+    def calculate_n_high_z(self) -> DataArray:
+        self.n_high_z = self._calculate_n_high_z()
+        return self.n_high_z
+
+    def _extrapolate_n_high_z(self) -> DataArray:
+        raise NotImplementedError(
+            f"{self.__class__} does not implement _extrapolate_n_high_z method"
+        )
+
+    def extrapolate_n_high_z(self) -> DataArray:
+        self.n_high_z = self._extrapolate_n_high_z()
+        return self.n_high_z
+
+    def _rescale_n_high_z(self) -> DataArray:
+        raise NotImplementedError(
+            f"{self.__class__} does not implement _rescale_n_high_z method"
+        )
+
+    def rescale_n_high_z(self) -> DataArray:
+        self.n_high_z = self._rescale_n_high_z()
+        return self.n_high_z
+
+    def _optimise_n_high_z(self) -> DataArray:
+        raise NotImplementedError(
+            f"{self.__class__} does not implement _optimise_n_high_z method"
+        )
+
+    def optimise_n_high_z(self) -> DataArray:
+        self.n_high_z = self._optimise_n_high_z()
```

Comment:
As much as it pains me we should probably stick to [oxford spelling](https://en.wikipedia.org/wiki/Oxford_spelling), or at least be consistent.